sign/bls/bls.go

```diff
@@ -1,6 +1,13 @@
 // Package bls implements the Boneh-Lynn-Shacham (BLS) signature scheme which
 // was introduced in the paper "Short Signatures from the Weil Pairing". BLS
 // requires pairing-based cryptography.
+//
+// Deprecated: This version is vulnerable to rogue public-key attack and the
+// new version of the protocol should be used to make sure a signature
+// aggregate cannot be verified by a forged key. You can find the protocol
+// in kyber/sign/bls2
+//
+// See the paper: https://crypto.stanford.edu/~dabo/pubs/papers/BLSmultisig.html
```

Comment:
Hmm, I think this comment should be reformulated. Using regular BLS signing without aggregation is perfectly fine and secure.  Meaning we only become a problem when regular BLS signatures are aggregated. I would probably remove all of the BLS aggregation code from this package, only expose regular BLS signing and verification, and then also remove the "deprecated flag".

Reply:
Keep in mind we need to stay backwards compatible for the moment. Eventually we won't have two packages but one. That's only for v4 though.

Reply:
Okay, then perhaps make it clear that only the signature aggregation is broken/deprecated (and will be phased out in a future version of kyber). The regular BLS signing and verification is still fine (and thus will remain in kyber).